packages/experimental/src/Machine.ts

```diff
@@ -0,0 +1,719 @@
+    self: M,
+    ...tags: Tags
+  ): Machine.ExcludeTag<M, Tags[number]> => ({ ...self, internalTags: [...self.internalTags, ...tags] }) as any
+)
+
+/**
+ * @since 1.0.0
+ * @category runtime
+ */
+export const boot = <
+  M extends Machine.Any
+>(
+  self: M,
+  ...[input, options]: [Machine.Input<M>] extends [void] ? [
+      input?: Machine.Input<M>,
+      options?: { readonly snapshot?: Machine.State<M>; readonly forever: boolean }
+    ] :
+    [
+      input: Machine.Input<M>,
+      options?: { readonly snapshot?: Machine.State<M>; readonly forever: boolean }
+    ]
+): Effect.Effect<Booted<M>, Machine.InitError<M>, Machine.Context<M> | Scope.Scope> => {
+  const procedureMap: Record<string, Procedure.Procedure<any, Machine.State<M>, never>> = Object.fromEntries(
+    self.procedures.map((p) => [p.tag, p])
+  )
+  const externalProcedures: Record<string, Procedure.Procedure<any, Machine.State<M>, never>> = Object.fromEntries(
+    self.procedures.filter((p) => !self.internalTags.includes(p.tag)).map((p) => [p.tag, p])
+  )
+
+  return Effect.gen(function*(_) {
+    const requests = yield* _(Queue.unbounded<[Schema.TaggedRequest.Any, Deferred.Deferred<any, any>]>())
+    const pubsub = yield* _(Effect.acquireRelease(
+      PubSub.unbounded<Machine.State<M>>(),
+      PubSub.shutdown
+    ))
+    const fiberSet = yield* _(FiberSet.make<any, never>())
+
+    let [state, initContext] = yield* _(self.initialize(input as Machine.Input<M>, options?.snapshot))
+
+    const send = <R extends Machine.Requests<M>>(request: R) =>
+      Effect.flatMap(
+        Deferred.make<Request.Success<R>, Request.Error<R>>(),
+        (deferred) =>
+          Queue.offer(requests, [request, deferred]).pipe(
+            Effect.zipRight(Deferred.await(deferred)),
+            Effect.onInterrupt(() => Deferred.interrupt(deferred))
+          )
+      )
+
+    const sendExternal = <R extends Machine.Requests<M>>(request: R) =>
+      request._tag in externalProcedures ? send(request) : Effect.die(`Unknown request: ${request._tag}`)
+
+    const publishState = (newState: Machine.State<M>) => {
+      if (state !== newState) {
+        state = newState
+        return PubSub.publish(pubsub, newState)
+      }
+      return Effect.unit
+    }
+
+    const fork = <A, E, R>(effect: Effect.Effect<A, E, R>) => FiberSet.run(fiberSet, Effect.orDie(effect))
+
+    const forkWith = <A, E, R>(state: Machine.State<M>, effect: Effect.Effect<A, E, R>) =>
+      Effect.as(FiberSet.run(fiberSet, Effect.orDie(effect)), [void 0, state] as const)
+
+    const forkWithState = <A, E, R>(effect: Effect.Effect<A, E, R>) =>
+      Effect.map(FiberSet.run(fiberSet, Effect.orDie(effect)), () => [void 0, state] as const)
+
+    const context: Procedure.Procedure.Context<Machine.Requests<M>, Machine.State<M>> = {
+      send,
+      fork,
+      forkWith,
+      forkWithState
+    }
+
+    const run = pipe(
```

Comment:
Move initialise into run

Expose .run instead of .join? So you can add your own retry policies.

Reply:
Also lift defects into a MachineError so retry policies are effective.